docs/guides/nextauth-migration.mdx

```diff
@@ -0,0 +1,253 @@
+You can simply replace the existing UI with Clerk's fully pre-built `<SignIn>` and `<SignUp>` components.
+
+```tsx
+// App Router   - /app/sign-in/[[...sign-in]]/page.tsx
+// Pages Router - /pages/sign-in/[[...index]].tsx
+
+import { SignIn } from "@clerk/nextjs"
+ 
+export default function Page() {
+  return <SignIn />;
+}
+```
+
+For fully custom UI, replace the existing functionality with [appropriate custom flows](https://clerk.com/docs/custom-flows/overview#custom-flows). With Custom flows, you leverage Clerk's React hooks `useSignIn` and `useSignUp` to build fully custom UI that handles authentication with email and password, OAuth, magic links, OTP etc.
+
+(maybe a small example?)
+
+## Protecting your app
+
+**Provider**
+
+Replace `<SessionProvider session={session}>` with `<ClerkProvider {...pageProps}>` (or just `<ClerkProvider>` for App Router)
+
+```tsx
+// With NextAuth.js
+
+import { SessionProvider } from "next-auth/react"
+
+export default function App({
+  Component,
+  pageProps: { session, ...pageProps },
+}) {
+  return (
+    <SessionProvider session={session}>
+      <Component {...pageProps} />
+    </SessionProvider>
+  )
+}
+```
+
+```tsx
+// With Clerk
+
+import { ClerkProvider } from "@clerk/nextjs";
+
+export default function App({ Component, pageProps }) {
+  return (
+    <ClerkProvider {...pageProps}>
+      <Component {...pageProps} />
+    </ClerkProvider>
+  );
+}
+```
+
+**Middleware**
+
+Replace NextAuth middleware with clerkMiddleware, use `protectedRoutes` as needed.
+
+```tsx
+// With NextAuth.js
+
+export { default } from "next-auth/middleware"
+
+export const config = { matcher: [/** private routes */] }
+```
+
+```tsx
+// With Clerk
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
+import { NextResponse } from "next/server";
+
+// opt into protection by adding routes to this array.
+const isProtectedRoute = createRouteMatcher([/** private routes */]);
+
+export default clerkMiddleware((auth, req) => {
+  // Check if the user visiting a protected route is logged in
+  // If not, redirect the user to the sign-in route
+  if (!auth().userId && isProtectedRoute(req)) {
+    return auth().redirectToSignIn();
+  }
+
+  // User isn't logged in and the route is public -- let them visit it
+  return NextResponse.next();
+});
+
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+NextAuth's middleware always rejects unauthorized requests, and you might be using the Next.js middleware config to protect specific private routes. Clerk's middleware can run for your entire application and gives you fine-grained control over handling the authenticated state. In this snippet, `createRouteMatcher` is used to define routes that should be protected, and the callback function checks for protected routes and redirects the user to a sign in page.
+
+**Rendering**
+
+To conditionally render UI when the user is signed in, wrap it with `<SignedIn>`
```

Comment:
```suggestion
To conditionally render UI when the user is signed in, wrap it with `[<SignedIn>](/docs/components/control/signed-in)`
```

Reply:
You applied the links, but I think the format I wrote them in is wrong and they are not links.